libc/include/llvm-libc-macros/sys-mman-macros.h

```diff
@@ -9,8 +9,40 @@
 #ifndef __LLVM_LIBC_MACROS_SYS_MMAN_MACROS_H
 #define __LLVM_LIBC_MACROS_SYS_MMAN_MACROS_H
 
-#ifdef __linux__
-#include "linux/sys-mman-macros.h"
+// Use definitions from <linux/mman.h> to dispatch arch-specific flag values.
+// For example, MCL_CURRENT/MCL_FUTURE/MCL_ONFAULT are different on different
+// architectures.
+#if __has_include(<linux/mman.h>)
+#include <linux/mman.h>
+#else
+#error "cannot use <sys/mman.h> without proper system headers."
+#endif
+
+// Some posix standard flags may not be defined in system headers.
```

Comment:
are these still necessary?

Reply:
Yes, `MAP_FAILED` and `POSIX_*` are not provided in UAPI.

Reply:
https://elixir.bootlin.com/linux/latest/A/ident/MAP_FAILED